### PR TITLE
Fix issue with Masking layer with Tensor as `mask_value`

### LINKED ISSUE #20706: Loaded Keras Model Throws Error While Predicting (Likely Issues with Masking)
I am currently developing and testing a RNN that relies upon a large amount of data for training, and so have attempted to separate my training and testing files. I have one file where I create, train, and save a tensorflow.keras model to a file 'model.keras' I then load this model in another file and predict some values, but get the following error: Failed to convert elements of {'class_name': '__tensor__', 'config': {'dtype': 'float64', 'value': [0.0, 0.0, 0.0, 0.0]}} to Tensor. Consider casting elements to a supported type. See https://www.tensorflow.org/api_docs/python/tf/dtypes for supported TF dtypes

By the way, I have tried running model.predict with this exact same data in the file where I train the model, and it works smoothly. The model loading must be the problem, not the data used to predict.

This mysterious float64 tensor is the value I passed into the masking layer. I don't understand why keras is unable to recognize this JSON object as a Tensor and apply the masking operation as such. I have included snippets of my code below, edited for clarity and brevity:

model_generation.py:
```
# Create model

model = tf.keras.Sequential([
    tf.keras.layers.Input((352, 4)),
    tf.keras.layers.Masking(mask_value=tf.convert_to_tensor(np.array([0.0, 0.0, 0.0, 0.0]))),
    tf.keras.layers.GRU(50, return_sequences=True, activation='tanh'),
    tf.keras.layers.Dropout(0.2),
    tf.keras.layers.GRU(50,activation='tanh'),
    tf.keras.layers.Dropout(0.2),
    tf.keras.layers.Dense(units=1, activation='sigmoid')])

# Compile Model...
# Train Model...
model.save('model.keras')

model.predict(data) # Line works here
```
model_testing.py
```
model = tf.keras.models.load_model('model.keras')

model.predict(data) # this line generates the error
```

I have tried to re-load the model in the `model_generation.py` file and I get the exact same issue.

### PR DESCRIPTION
**Pull request by Surya2k1**
Currently , the Masking layer accepting Tensor also for argument `mask_value` . After saving and loading the model the `mask_value` no longer a tensor. Hence the issue. The issue with be an model which have Masking layer and uses a Tensor as `mask_value`.

Either the Masking layer should not allow Tensors for `mask_value` or while reloading the model we need to take care of this explicitly.

I am proposing to handle `mask_value` explicitly during reloading the model.

Might fix #20706

### CODE REVIEW

File: keras/src/layers/core/masking.py [hunk 1]
```diff
@@ -45,6 +45,8 @@ class Masking(Layer):
 
     def __init__(self, mask_value=0.0, **kwargs):
         super().__init__(**kwargs)
+        if isinstance(mask_value, dict) and mask_value.get("config", None):
+            mask_value = mask_value["config"]["value"]
```
Comment (fchollet) [inline keras/src/layers/core/masking.py:49]: You should be able to do instead `mask_value = serialization_lib.deserialize_keras_object(mask_value)`, it would be more general

**Review by fchollet — COMMENTED** [review]
Thanks for the PR!

You can fix the code format by running `sh shell/format.sh`

File: keras/src/layers/core/masking.py (hunk 1 above)
Comment (Surya2k1) [reply keras/src/layers/core/masking.py]: Thanks for info. Done the changes as suggested.

File: keras/src/layers/core/masking.py [hunk 2]
```diff
@@ -45,6 +46,9 @@ class Masking(Layer):
 
     def __init__(self, mask_value=0.0, **kwargs):
         super().__init__(**kwargs)
+        # `mask_value` can be a serialized tensor, hence verify it
+        if isinstance(mask_value, dict) and mask_value.get("config", None):
```
Comment (fchollet) [inline keras/src/layers/core/masking.py:50]: You can actually remove this line! `deserialize_keras_object` should work in all circumstances.
